src/main/java/org/broadinstitute/hellbender/transformers/DUSTReadTransformer.java

```diff
@@ -0,0 +1,230 @@
+package org.broadinstitute.hellbender.transformers;
+
+import htsjdk.samtools.Cigar;
+import htsjdk.samtools.util.Tuple;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.broadinstitute.hellbender.cmdline.Argument;
+import org.broadinstitute.hellbender.tools.ClipReads;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import htsjdk.samtools.SAMUtils;
+
+import java.util.*;
+
+//Masks low-complexity sequences with 'N' and base qualities with '#'
+
+public class DUSTReadTransformer implements ReadTransformer {
+    public static final long serialVersionUID = 1L;
+    private final Logger logger = LogManager.getLogger(DUSTReadTransformer.class);
+
+    @Argument(fullName="Mask PHRED score", shortName="dustPhred", optional=true)
+    public short MASK_PHRED = 15;
+
+    @Argument(fullName="Window size", shortName="dustW", optional=true)
+    public int W = 64;
+
+    @Argument(fullName="Score threshold", shortName="dustT", optional=true)
+    public float T = 20;
+
+    private static final int NUM_TRIPLETS = 64;
+
+    private class PStruct {
+        public int start = 0;
+        public int finish = 0;
+        public float score = 0;
+    }
+
+    @Override
+    public GATKRead apply(final GATKRead read) {
+
+        byte[] q = read.getBases();
+
+        //Find convert to 2-bit and assign random bases to N's
+        //TODO: reject reads with too many N's
+        Random rng = new Random();
+        rng.setSeed(0);
+        for (int i = 0; i < q.length; i++) {
+            if (q[i] == 'N') {
+                q[i] = (byte)rng.nextInt(4);
+            } else {
+                if (q[i] == 'A' || q[i] == 'a') {
+                    q[i] = 0x0;
+                } else if (q[i] == 'T' || q[i] == 't') {
+                    q[i] = 0x1;
+                } else if (q[i] == 'C' || q[i] == 'c') {
+                    q[i] = 0x2;
+                } else if (q[i] == 'G' || q[i] == 'g') {
+                    q[i] = 0x3;
+                } else {
+                    logger.warn("Invalid base " + (char) q[i] + " in read " + read.getName() + ", assigning to random base.");
+                    q[i] = (byte)rng.nextInt(4);
+                }
+            }
+        }
+
+        //Start main algorithm
+        List<Tuple<Integer,Integer>> res = new ArrayList<>();
+        LinkedList<PStruct> P = new LinkedList<>();
```

Comment:
I think it would add a lot to the readability/maintainability of this algorithm if you could assign some more descriptive names to variables and structures. (I realize you probably based this on someone else's implementation or description). P, PStruct, rv, rw, cv, cw, and L are all pretty impenetrable names. Based on your understanding of the algorithm, do you think you could do some renaming here? Failing that, a link to somewhere that explains the algorithm and would let people decipher it would definitely help.